File: src/service/to/chat-session.ts
import ChatMessage from './chat-message';
import IdentityUtil from '../../util/identity-util';
import Persona, {DEFAULT_PERSONA} from './persona';

class ChatSession {
  sessionId: string;
  systemMessages: Array<ChatMessage> = [];
  messages: Array<ChatMessage> = [];
  processedMessages: Array<ChatMessage> = [];
  persona: Persona;

  constructor() {
    this.sessionId = IdentityUtil.generateUUID();
    this.systemMessages = [];
    this.messages = [];
    this.processedMessages = [];
    this.persona = DEFAULT_PERSONA;
  }

}

export default ChatSession;

File: src/main/java/com/talkforgeai/talkforgeaiserver/dto/ChatSession.java
package com.talkforgeai.talkforgeaiserver.dto;

import com.theokanning.openai.completion.chat.ChatMessage;

import java.util.ArrayList;
import java.util.List;

public record ChatSession(
        String sessionId,
        List<ChatMessage> systemMessages,
        List<ChatMessage> messages,
        List<ChatMessage> processedMessages) {

    public ChatSession(String sessionId) {
        this(sessionId, new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
    }

}
